### DISCUSSION TITLE
Using the gopro to monitor sealife

### DISCUSSION THREAD

**Discussion by NOGP1** [gopro/labs#935] [Q&A] [answered]
I am currently working on a project where I need the gopro to take photos of sea animals. I have made it work before, but now im really struggling to get it to work again. 

I have a timer that delivers power to the camera for 20 seconds every 30 min. I need the gopro to take a photo when power is delivered every 30 min. 

I get it to turn on and start capturing, but only video... I have used: !MTUSB=1!R!MBOOT="!LRecOnUsb"!SAVERecOnUsb=>u!S~!10E!R and several codes for photo, and ive used !R to try and save the command, but everytime the camera turn on again its back to video... Like this on, and many others: mPfW0r0tcNwAi2M1x-.5sLq0oV1B1D4L0!R!MWAKE=2!MTUSB=1!MBOOT="!LRecOnUsb"!SAVERecOnUsb

The photo needs to be Wide, ISO min-max (100-200), standard, natural, LOW sharpness. 

Is this something you can help with?

Thanks!

**Comment by dnewman-gpsw** [comment]
Which camera, which firmware?  <- always required.

This is will not work:
mPfW0r0tcNwAi2M1x-.5sLq0oV1B1D4L0**!R**!MWAKE=2!MTUSB=1!MBOOT="!LRecOnUsb"!SAVERecOnUsb

Never use !R this way, it causing the repeat to happen for all you settings continuously.

This too is dangerous
>u!S~!10E!R

as pseudo code:
```
loop:
if(USB power)
{
   Start();
}
else
{
  Sleep(10);
  Stop();
}
goto loop;
```
This is weird, you can see it will not work.

**Reply by NOGP1** [reply]
Thanks, i am using a gopro 12 with the newest update. I got it to work using 
![this ](https://github.com/gopro/labs/assets/171022359/16283d8e-d0c9-4ef1-9acb-0e01c9beb0c9)

Two codes together... Dont know how, but atleast its working as it should

**Reply by dnewman-gpsw** [reply]
Those codes are very bad. The first could ultimately damage a camera. You have it repeatedly writing to NVM, which has it limited number of write cycles, many times per second.  Remove that !R.  The RecOnUsb script is poorly designed. You have have in a constant loop saying Start, Start, Start.... even if the camera is already recording.  You should be testing if the camera is recording before issuing a start command.

**Reply by NOGP1** [reply]
I see, thank you! I will remove the !R code, but what substitute do i have for RecOnUsb? I use SaveRecOnUsb to make it work on a smaller electrical current, Tusb=1 for flowthrough without battery, !MWAKE to turn camera on when power is delivered, Boot aswell, but without timercommand and reconusb, i have found that it doesnt start automatically. The camera will never take more than one picture every hour, but if you are saying it could damage the camera then I really need to fix this. It will be capturing for several weeks at the time. 

Any further help is appreciated :)

**Reply by NOGP1** [reply]
Further, it will just be used for photography, never filming. One photo taken every 30-120 min for 1 month.

**Reply by NOGP1** [reply]
so far its been capturing for 2 days, no hiccups, but i will turn it off now.

**Reply by dnewman-gpsw** [reply, marked as answer]
Only the first command was potentially dangerous, the other was just looks a bit silly, but as you are externally switching, it make some sense. Externally switching is completely unnecessary on HERO12.

**Reply by NOGP1** [reply]
Thank you so much! I agree, but its necessary because we have a timer that delivers power to the camera aswell as a light. when the camera turns on and takes a photo, the light turns on aswell. We need the room to otherwise be totally dark. 

Only way i figured how to make the light go off together with the camera.

**Reply by NOGP1** [reply]
Question: The commands that i have put in says nothing about the settings for photography; Sharpness, ISO, WB etc. Can i adjust these setting in camera/app and then read the QR codes? or will they go back to "standard"? 

Thank you

**Reply by dnewman-gpsw** [reply]
I've used the camera to trigger an external light. https://youtube.com/shorts/1wJh_XayB3w

If you set a photo mode first, and then charge sharpness ISO, WB etc. it effects that mode. Order of operations in important.

**Reply by NOGP1** [reply]
Fantastic! Thank you, i will look into it. This is our camera setup now. Working great using timers, forcing the camera off by removing its power supply. Would be nice to work on internal timelapse mode where the camera turns itself off. 

![Skjermbilde 2024-06-06 093415](https://github.com/gopro/labs/assets/171022359/e089e18e-5f52-4948-8992-8c4a230fa929)
